app/src/main/java/org/sopt/korailtalk/core/designsystem/theme/Type.kt

```diff
@@ -1,34 +1,162 @@
 package org.sopt.korailtalk.core.designsystem.theme
 
-import androidx.compose.material3.Typography
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.staticCompositionLocalOf
 import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
 import androidx.compose.ui.text.font.FontFamily
-import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.sp
+import org.sopt.korailtalk.R
 
-// Set of Material typography styles to start with
-val Typography = Typography(
-    bodyLarge = TextStyle(
-        fontFamily = FontFamily.Default,
-        fontWeight = FontWeight.Normal,
-        fontSize = 16.sp,
-        lineHeight = 24.sp,
-        letterSpacing = 0.5.sp
+object PretendardFont {
+    val Medium = FontFamily(Font(R.font.pretendard_medium))
+    val Regular = FontFamily(Font(R.font.pretendard_regular))
+    val SemiBold = FontFamily(Font(R.font.pretendard_semibold))
+
+}
+
+
+sealed interface TypographyTokens {
+    @Immutable
+    data class Headline(
+        val head1_m_30: TextStyle,
+        val head2_m_20: TextStyle,
+        val head3_sb_18: TextStyle,
+        val head4_m_18: TextStyle,
+        val head5_r_18: TextStyle
+    )
+
+    @Immutable
+    data class Subtitle(
+        val sub1_m_17: TextStyle,
+        val sub2_r_17: TextStyle,
+        val sub3_m_16: TextStyle
+    )
+
+    @Immutable
+    data class Body(
+        val body1_r_16: TextStyle,
+        val body2_m_15: TextStyle,
+        val body3_r_15: TextStyle,
+        val body4_m_14: TextStyle,
+        val body5_r_13: TextStyle
     )
-    /* Other default text styles to override
-    titleLarge = TextStyle(
-        fontFamily = FontFamily.Default,
-        fontWeight = FontWeight.Normal,
-        fontSize = 22.sp,
-        lineHeight = 28.sp,
-        letterSpacing = 0.sp
+
+    @Immutable
+    data class Cap(
+        val cap1_m_12: TextStyle,
+        val cap2_r_12: TextStyle
+    )
+}
+
+@Immutable
+data class KorailTalkTypography(
+    val headline: TypographyTokens.Headline,
+    val subtitle: TypographyTokens.Subtitle,
+    val body: TypographyTokens.Body,
+    val cap: TypographyTokens.Cap
+)
+
+
+val defaultKorailTalkTypography = KorailTalkTypography(
+    headline = TypographyTokens.Headline(
+        head1_m_30 = TextStyle(
+            fontFamily = PretendardFont.Medium,
+            fontSize = 30.sp,
+            lineHeight = 39.sp, //130%
+            letterSpacing = (-1.5).sp
+        ),
```

Comment:
저희 그 피그마 상으로 행간이 130, 자간이 -1.5로 고정인 것 같은데,
```
private fun KorailTalkTextStyle(
    fontFamily: FontFamily,
    fontSize: TextUnit,
    lineHeight: TextUnit = 1.3.em,
    letterSpacing = (-1.5).sp
): TextStyle = TextStyle(
    fontFamily = fontFamily,
    fontSize = fontSize,
    lineHeight = lineHeight,
    letterSpacing = letterSpacing,
    lineHeightStyle = LineHeightStyle(
        alignment = LineHeightStyle.Alignment.Center,
        trim = LineHeightStyle.Trim.None
    )
)
```
를 위에 선언해두고,

여기서

```suggestion
        head1_m_30 = KorailTalkTextStyle(
            fontFamily = PretendardFont.Medium,
            fontSize = 30.sp
        ),
```
식으로 두면서 중복을 줄여볼 수 있을 것 같아요:) (변환은 지피티가 잘 해줄 것 같습니다..)